### ISSUE TITLE
Incorrectly find "__getitem__" method not defined on type "Type[Enum]"

### ISSUE THREAD

**Issue by ironfoot** [microsoft/pyright#1055]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
pyright incorrectly finds bracket access is not defined for enum types.

**To Reproduce**

```python
import enum
from typing import Type

class Color(enum.Enum):
    RED = enum.auto()
    GREEN = enum.auto()

print(Color['RED'])  # This is fine.

color: Type[enum.Enum] = Color
print(color['RED'])  # This is fine as well.

def test_fn(color: Type[enum.Enum]):
    print(color['RED'])  # "__getitem__" method not defined on type "Type[Enum]"
```

**Expected behavior**
No error.

**VS Code extension or command-line**
Pylance VS Code extension version: 2020.9.5

**Comment by erictraut** [comment]
Thanks for reporting this. It will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.74, which I just published. It will also be included in the next release of Pylance.
